### ISSUE TITLE
Annotations on java parameters do not appear in Documentable

### ISSUE THREAD

**Issue by owengray-google** [Kotlin/dokka#1694]
**Describe the bug**

Tl;dr: public void foo(@Nullable Integer arg1, @NonNull Integer arg2) cannot be handled in dokka: annotations on parameters don't work in java. Nullability is fairly important, and parameters seem to be the most common use case.

In our repo, we have the following two tests, which pass with the NOTE (which may be a bug in dokka as well):
```
    @Test
    fun `Method component has annotation and value in 4x Kotlin and Java`() {
        val annotationsK = """
            |annotation class Hello(val bar: String)
            |@Hello("abc")
            |@Hello(bar = "baz")
            |fun foo() = Unit
        """.render().functionAnnotations()
        val annotationsJ = """
            |@Retention(RetentionPolicy.RUNTIME)
            |@Target(ElementType.METHOD)
            |public @interface Hello {
            |    public String bar() default "";
            |}
            |@Hello("abc")
            |@Hello(bar = "baz")
            |public void foo() {}
        """.render(java = true).functionAnnotations()

        for (annotations in listOf(annotationsK, annotationsJ)) {
            val annotationOne = annotations.components().first()
            val parameterOne = annotationOne.data.parameters.item()
            val annotationTwo = annotations.components().last()
            val parameterTwo = annotationTwo.data.parameters.item()

            // NOTE: "value" in java does not match "bar" in kotlin
            if (annotations == annotationsK) assertThat(parameterOne.name).isEqualTo("bar")
            else assertThat(parameterOne.name).isEqualTo("value")
            assertThat(parameterOne.value).isEqualTo("\"abc\"")
            assertThat(parameterTwo.name).isEqualTo("bar")
            assertThat(parameterTwo.value).isEqualTo("\"baz\"")
        }
    }
```
```
    @Test
    fun `Property component has annotation and value in 4x Kotlin and Java`() {
        val annotationsK = """
            |annotation class Hello(val bar: String)
            |@Hello("abc")
            |@Hello(bar = "baz")
            |val foo: String = "foofoo"
        """.render().property()!!.annotations()
        val annotationsJ = """
            |@Retention(RetentionPolicy.RUNTIME)
            |@Target(ElementType.FIELD)
            |public @interface Hello {
            |    public String bar() default "";
            |}
            |@Hello("abc")
            |@Hello(bar = "baz")
            |public String foo = "foofoo"
        """.render(java = true).property()!!.annotations()

        for (annotations in listOf(annotationsK, annotationsJ)) {
            val annotationOne = annotations.components().first()
            val parameterOne = annotationOne.data.parameters.item()
            val annotationTwo = annotations.components().last()
            val parameterTwo = annotationTwo.data.parameters.item()

            // NOTE: "value" in java does not match "bar" in kotlin
            if (annotations == annotationsK) assertThat(parameterOne.name).isEqualTo("bar")
            else assertThat(parameterOne.name).isEqualTo("value")
            assertThat(parameterOne.value).isEqualTo("\"abc\"")
            assertThat(parameterTwo.name).isEqualTo("bar")
            assertThat(parameterTwo.value).isEqualTo("\"baz\"")
        }
    }
```
And we have the following two tests, which fail in the java version, because the DParameter has no Annotations when coming from Java.
```
    @Test
    fun `Parameter component has annotation and value in 4x Kotlin and Java`() {
        val annotationsK = """
            |annotation class Hello(val bar: String)
            |fun foo(@Hello("abc") @Hello(bar = "baz") arg: String) = Unit
        """.render().function()!!.parameters.single().annotations()
        val annotationsJ = """
            |@Retention(RetentionPolicy.RUNTIME)
            |@Target(ElementType.PARAMETER)
            |public @interface Hello {
            |    public String bar() default "";
            |}
            |public void foo(@Hello("abc") @Hello(bar = "baz") String arg)
        """.render(java = true).function()!!.parameters.single().annotations()

        for (annotations in listOf(annotationsK/*, annotationsJ*/)) {
            val annotationOne = annotations.components().first()
            val parameterOne = annotationOne.data.parameters.item()
            val annotationTwo = annotations.components().last()
            val parameterTwo = annotationTwo.data.parameters.item()

            if (annotations == annotationsK) assertThat(parameterOne.name).isEqualTo("bar")
            else assertThat(parameterOne.name).isEqualTo("value")
            assertThat(parameterOne.value).isEqualTo("\"abc\"")
            assertThat(parameterTwo.name).isEqualTo("bar")
            assertThat(parameterTwo.value).isEqualTo("\"baz\"")
        }
    }
```
```
    @Test
    fun `Type parameter component has annotation and value in 4x Kotlin and Java`() {
        val annotationsK = """
            |annotation class Hello(val bar: String)
            |fun <@Hello("abc") @Hello(bar = "baz") T> foo(arg: String): List<T>
        """.render().function()!!.generics.single().annotations()
        val annotationsJ = """
            |@Retention(RetentionPolicy.RUNTIME)
            |@Target(ElementType.TYPE_PARAMETER)
            |public @interface Hello {
            |    public String bar() default "";
            |}
            |public <@Hello("abc") @Hello(bar = "baz") T> List<T> foo()
        """.render(java = true).function()!!.generics.single().annotations()

        for (annotations in listOf(annotationsK/*, annotationsJ*/)) {
            val annotationOne = annotations.components().first()
            val parameterOne = annotationOne.data.parameters.item()
            val annotationTwo = annotations.components().last()
            val parameterTwo = annotationTwo.data.parameters.item()

            if (annotations == annotationsK) assertThat(parameterOne.name).isEqualTo("bar")
            else assertThat(parameterOne.name).isEqualTo("value")
            assertThat(parameterOne.value).isEqualTo("\"abc\"")
            assertThat(parameterTwo.name).isEqualTo("bar")
            assertThat(parameterTwo.value).isEqualTo("\"baz\"")
        }
    }
```
I believe this bug may also occur with the following input, but I don't have a test for it:
```
        val annotationsK = """
            |annotation class Hello(val bar: String)
            |fun <T : @Hello("abc") @Hello(bar = "baz") String> foo(arg: String): List<T>
        """.render()
        val annotationsJ = """
            |@Retention(RetentionPolicy.RUNTIME)
            |@Target(ElementType.TYPE_PARAMETER)
            |public @interface Hello {
            |    public String bar() default "";
            |}
            |public <T extends @Hello("abc") @Hello(bar = "baz") String> List<T> foo()
        """.render(java = true)
```

**Screenshots**
<img width="1201" alt="Screen Shot 2021-01-13 at 3 32 24 PM" src="https://user-images.githubusercontent.com/56644404/104507370-a1164e80-55b4-11eb-9ab1-cece3d9cd3c4.png">

**Dokka Configuration**
Operating system: macOS/Windows/Linux
Build tool: Gradle v6.5.1
dokkaVersion = "1.4.20-dev-36"
Google's dokka-devsite-plugin project
